transformer_engine/pytorch/csrc/extensions/gemm.cpp

```diff
@@ -175,8 +175,14 @@ std::vector<py::object> gemm(py::handle A, bool transa, py::handle B, bool trans
   const int sm_count = transformer_engine::cuda::sm_count(device_id);
   int num_math_sms = sm_count - transformer_engine::getenv<int>("NVTE_EXT_MARGIN_SM", sm_count);
 
+  std::vector<std::optional<at::Tensor>> original_scale_inverses_list;
```

Comment:
We're actually holding onto the swizzled scales, not the original:
```suggestion
  std::vector<std::optional<at::Tensor>> swizzled_scale_inverses_list;
```
The purpose of this list is also a little unintuitive since we never use it. Perhaps we should add a comment that it is keeping PyTorch tensors alive.

We should make similar changes in `te_general_grouped_gemm`.

Reply:
Updated.